### PR TITLE
Update `PodSpec.Overhead` broken links

### LINKED ISSUE #100244: The document of Overhead mentioned in the note cannot find
#### What happened:
/sig docs
The [document of Overhead mentioned](https://github.com/kubernetes/kubernetes/blob/9657dd77a3563ffc7dbd061d51f34f714b1f8d5c/pkg/apis/core/types.go#L2851) `https://git.k8s.io/enhancements/keps/sig-node/20190226-pod-overhead.md` in the note cannot find 

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Update `PodSpec.Overhead` broken links

Refer to the[ KEP](https://git.k8s.io/enhancements/keps/sig-node/688-pod-overhead/README.md), [This field is beta-level as of Kubernetes v1.18](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/688-pod-overhead/README.md#version-118)

#### Which issue(s) this PR fixes:

Fixes #100244

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -2848,7 +2848,7 @@ type PodSpec struct {
 	// The RuntimeClass admission controller will reject Pod create requests which have the overhead already
 	// set. If RuntimeClass is configured and selected in the PodSpec, Overhead will be set to the value
 	// defined in the corresponding RuntimeClass, otherwise it will remain unset and treated as zero.
-	// More info: https://git.k8s.io/enhancements/keps/sig-node/20190226-pod-overhead.md
+	// More info: https://git.k8s.io/enhancements/keps/sig-node/688-pod-overhead/README.md
 	// This field is alpha-level as of Kubernetes v1.16, and is only honored by servers that enable the PodOverhead feature.
```
Comment (liggitt) [inline pkg/apis/core/types.go:2852]: thanks for fixing this. while we're here, the field is now beta-level as of 1.18 (applies everywhere)

**Review by liggitt — COMMENTED** [review]
thanks for the fix, noted one update needed to indicate correct beta status.

no API review is required for a doc fix like this, but thanks for checking.

File: pkg/apis/core/types.go (hunk 1 above)
Comment (mengjiao-liu) [reply pkg/apis/core/types.go:2852]: Thanks for your reminding. I will update the level description of this field

**Review by ehashman — COMMENTED** [review]
/priority backlog
/triage accepted
/kind cleanup

**Pull request merged**